client/ayon_deadline/lib.py

```diff
@@ -43,3 +47,369 @@ def get_instance_job_envs(instance) -> "dict[str, str]":
+
+
+@dataclass
+class AYONDeadlineJobInfo(DeadlineJobInfo):
+    """Contains additional AYON variables from Settings for internal logic."""
+
+    # AYON custom fields used for Settings
+    UsePublished: Optional[bool] = field(default=None)
+    UseAssetDependencies: Optional[bool] = field(default=None)
+    UseWorkfileDependency: Optional[bool] = field(default=None)
+
+    def serialize(self):
+        """Return all data serialized as dictionary.
+
+        Returns:
+            OrderedDict: all serialized data.
+
+    """
+        def filter_data(a, v):
+            if isinstance(v, (DeadlineIndexedVar, DeadlineKeyValueVar)):
+                return False
+            if v is None:
+                return False
+            return True
+
+        serialized = asdict(self)
+        serialized = {
+            k: v for k, v in serialized.items()
+            if filter_data(k, v)
+        }
+
+        # Custom serialize these attributes
+        for attribute in [
+            self.EnvironmentKeyValue,
+            self.ExtraInfo,
+            self.ExtraInfoKeyValue,
+            self.TaskExtraInfoName,
+            self.OutputFilename,
+            self.OutputFilenameTile,
+            self.OutputDirectory,
+            self.AssetDependency
+        ]:
+            serialized.update(attribute.serialize())
+
+        return serialized
+
+    @classmethod
+    def from_dict(cls, data: Dict) -> 'AYONDeadlineJobInfo':
+
+        implemented_field_values = {
+            "ChunkSize": data["chunk_size"],
+            "Priority": data["priority"],
+            "MachineLimit": data["machine_limit"],
+            "ConcurrentTasks": data["concurrent_tasks"],
+            "Frames": data["frames"],
+            "Group": cls._sanitize(data["group"]),
+            "Pool": cls._sanitize(data["primary_pool"]),
+            "SecondaryPool": cls._sanitize(data["secondary_pool"]),
+
+            # fields needed for logic, values unavailable during collection
+            "UsePublished": data["use_published"],
+            "UseAssetDependencies": data["use_asset_dependencies"],
+            "UseWorkfileDependency": data["use_workfile_dependency"]
+        }
+
+        return cls(**implemented_field_values)
+
+    def add_render_job_env_var(self):
+        """Add required env vars for valid render job submission."""
+        for key, value in get_ayon_render_job_envs().items():
+            self.EnvironmentKeyValue[key] = value
+
+    def add_instance_job_env_vars(self, instance):
+        """Add all job environments as specified on the instance and context
+
+        Any instance `job_env` vars will override the context `job_env` vars.
+        """
+        for key, value in get_instance_job_envs(instance).items():
+            self.EnvironmentKeyValue[key] = value
+
+    def to_json(self) -> str:
+        """Serialize the dataclass instance to a JSON string."""
+        return json.dumps(asdict(self))
+
+    @classmethod
+    def _sanitize(cls, value) -> str:
```

Comment:
This can also be a staticmethod

```suggestion
    @staticmethod
    def _sanitize(value) -> str:
```

Reply:
BTW I don't know if this is new thing, but why are we using value `"none"` instead of `""`?